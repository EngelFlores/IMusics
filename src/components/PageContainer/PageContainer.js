import React from 'react';
import './PageContainer.css';
import Spinner from '../Spinner/Spinner';

const PageContainer = ({ children, loaded = true }) => {
  return (
    <div
      className={'page-container'}
      data-testid="pageContainer"
    >
      {loaded ? children : <Spinner />}
    </div>
  );
};

export default PageContainer;
