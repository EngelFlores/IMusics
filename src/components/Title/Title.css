.title {
  margin-top: 120px;
  margin-left: 20px;
  margin-right: 20px;
  display: flex;
  justify-content: space-between
}

h1,
a {
  font-size: 22px;
  font-weight: 600
}
