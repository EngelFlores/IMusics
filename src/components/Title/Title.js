import React from 'react';
import { Link } from 'react-router-dom';

import './Title.css'

// eslint-disable-next-line no-undef
const Title = ({ title, shouldShow }) => {
  return (
    <div className="title" data-testid="title">
      <h1>{title}</h1>
      {shouldShow && <Link to={'/artists'}>Voltar</Link>}
    </div>
  );
}

export { Title }
