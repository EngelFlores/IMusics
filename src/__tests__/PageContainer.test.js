import 'jest-dom/extend-expect'
import { cleanup, render, waitForElement } from 'react-testing-library'
import PageContainer from '../components/PageContainer/PageContainer'
import { Title } from '../components/Title/Title'
import React from 'react'

describe('Spinner', () => {
  afterEach(cleanup)

  test('Return spinner when loaded is true', async () => {
    const { getByTestId } = render(<PageContainer loaded={true} />)
    const pageContainer = await waitForElement(() => getByTestId('pageContainer'))

    expect(pageContainer).toMatchSnapshot()
  })

  test('Return children when loaded is false', async () => {
    const { getByTestId } = render(<PageContainer loaded={false} children={Title}/>)
    const pageContainer = await waitForElement(() => getByTestId('pageContainer'))

    expect(pageContainer).toMatchSnapshot()
  })

})
