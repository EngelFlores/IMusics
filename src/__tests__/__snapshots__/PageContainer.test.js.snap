// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Return children when loaded is false 1`] = `
<div
  class="page-container"
  data-testid="pageContainer"
>
  <div
    class="spinner"
  >
    <img
      alt="loading"
      class="spinner__needle"
      data-testid="needle"
      src="needle.svg"
    />
    <img
      alt="loading"
      class="spinner__record"
      data-testid="vinyl"
      src="vinyl.svg"
    />
  </div>
</div>
`;

exports[`Spinner Return spinner when loaded is true 1`] = `
<div
  class="page-container"
  data-testid="pageContainer"
/>
`;
